### DISCUSSION TITLE
Consult breaks functionality in rename-file dialog

### DISCUSSION THREAD

**Discussion by lagunoff** [minad/consult#1062] [Q&A] [answered]
I installed consult from emacs package-list-packages interface and been enjoying using it until I noticed I cannot rename a file into a shorter version of its name because consult would automatically chose first suggested version which is the old version of the file name. 

```
consult-20240710.2028
GNU Emacs 29.3

; no extra configuration, only config that could be relevant is these lines
(marginalia-mode 1)
(vertico-mode 1)
```

Here is me trying to rename `/tmp/foo.txt` -> `/tmp/foo`. When I press `Enter` it says `Cannot move to same file: /tmp/foo.txt`
![image](https://github.com/user-attachments/assets/c7091ee2-ab09-4ff8-8f50-476ff13e44e2)

**Comment by lagunoff** [comment]
I'd say ideally another set of behaviour should be implemented when rename-file asks for the second argument. I don't know much about consult internals but I imagine this could be tricky even if it's possible to determine which argument is currently requested. 

Another low-effort solution that would be sufficient for me personally is to disable consult specifically inside rename-file function. I would really appreciate if anyone can provide a configuration that can turn off consult inside rename-file.

**Reply by minad** [reply, marked as answer]
(The question is related to Vertico, not Consult) This question has been asked many times, please see https://github.com/minad/vertico?tab=readme-ov-file#key-bindings, specifically M-RET.

**Reply by lagunoff** [reply]
Hitting M-RET instead of just RET when I'm done typing resolved the issue. I guess I should've read the docs before posting. Thanks anyway for pointing out the solution!
